src/sources/exec/mod.rs

```diff
@@ -0,0 +1,789 @@
+            Error::new(ErrorKind::Other, "Unable to take stderr of spawned process")
+        })?;
+
+        // Create stderr async reader
+        let stderr = stderr.allow_read_until(shutdown.clone().map(|_| ()));
+        let stderr_reader = BufReader::new(stderr);
+
+        spawn_reader_thread(
+            stderr_reader,
+            config.event_per_line,
+            config.maximum_buffer_size_bytes,
+            STDERR,
+            sender.clone(),
+        );
+    }
+
+    let stdout = child
+        .stdout
+        .take()
+        .ok_or_else(|| Error::new(ErrorKind::Other, "Unable to take stdout of spawned process"))?;
+
+    // Create stdout async reader
+    let stdout = stdout.allow_read_until(shutdown.clone().map(|_| ()));
+    let stdout_reader = BufReader::new(stdout);
+
+    let pid = child.id();
+
+    spawn_reader_thread(
+        stdout_reader,
+        config.event_per_line,
+        config.maximum_buffer_size_bytes,
+        STDOUT,
+        sender,
+    );
+
+    while let Some((line, stream)) = receiver.recv().await {
+        let event = create_event(&config, &hostname, line, &Some(stream.to_string()), pid);
+
+        let _ = out
+            .send(event)
+            .await
+            .map_err(|_: crate::pipeline::ClosedError| {
+                error!(message = "Failed to forward events; downstream is closed.");
+            });
+    }
+
+    let elapsed = start.elapsed();
+
+    let result = match child.try_wait() {
+        Ok(Some(exit_status)) => {
+            handle_exit_status(&config, exit_status.code(), elapsed);
+            Ok(Some(exit_status))
+        }
+        Ok(None) => {
+            handle_exit_status(&config, None, elapsed);
+            Ok(None)
+        }
+        Err(error) => {
+            error!(message = "Unable to obtain exit status.", %error);
+
+            handle_exit_status(&config, None, elapsed);
+            Ok(None)
+        }
+    };
+
+    debug!("Finished command run.");
+    let _ = out.flush().await;
+
+    result
+}
+
+fn handle_exit_status(config: &ExecConfig, exit_status: Option<i32>, exec_duration: Duration) {
+    emit!(ExecCommandExecuted {
+        command: config.command_line().as_str(),
+        exit_status,
+        exec_duration,
+    });
+}
+
+fn build_command(config: &ExecConfig) -> Command {
+    let command = &config.command[0];
+
+    let mut command = Command::new(command);
+
+    if config.command.len() > 1 {
+        command.args(&config.command[1..]);
+    };
+
+    command.kill_on_drop(true);
+
+    // Explicitly set the current dir if needed
+    if let Some(current_dir) = &config.working_directory {
+        command.current_dir(current_dir);
+    }
+
+    // Pipe our stdout to the process
+    command.stdout(std::process::Stdio::piped());
+
+    // Pipe stderr to the process if needed
+    if config.include_stderr {
+        command.stderr(std::process::Stdio::piped());
+    } else {
+        command.stderr(std::process::Stdio::null());
```

Comment:
I'm not sure this is wise, as it makes any error output invisible when we don't pull it in as input, but I'm not sure. @jszwedko?

Reply:
Hmm. In my ideal world:

* If `include_stderr` is `false` any stderr of the forked process is just dropped
* If `include_stderr` is `true` any stderr of the forked process is pushed as events

I think that is what is happening here, yes? I can test tomorrow.

Reply:
Yes, that is what I would expect from this code. If that matches your expectations then I am fine with it.

Reply:
Just pushed a fix for the cue format (I am not sure why `cue fmt ./docs/**/*.cue` didn't fix it.

Regarding include_stderr, if you guys decide that this configuration is not needed (i.e. we would always need stderr) I would be happy to remove it. Perhaps stderr could be filtered out by a transformation instead.

Reply:
👍 I tested and the behavior seems to be what I expected as mentioned in the above comment.